Bottom: c9523e4c09c02f7e5877579e101ef6260ff3f565
Top:    ff61539794a40a2f4809f311878c6cc0fdd13541
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-23 23:36:36 +0000

Refresh of work-in-progress-on-subcomposi

---

diff --git a/src/hid/gerber/gerber.c b/src/hid/gerber/gerber.c
index 7fc62f9..9dc6d2d 100644
--- a/src/hid/gerber/gerber.c
+++ b/src/hid/gerber/gerber.c
@@ -455,6 +455,9 @@ gerber_set_layer (const char *name, int group, int empty)
 	     && group <
 	     max_layer) ? PCB->LayerGroups.Entries[group][0] : group;
 
+  if (SL_TYPE (idx) == SL_FINISHED)
+    return 0;
+
   if (name == 0)
     name = PCB->Data->Layer[idx].Name;
 
diff --git a/src/hid/nelma/nelma.c b/src/hid/nelma/nelma.c
index d06193c..e548bc0 100644
--- a/src/hid/nelma/nelma.c
+++ b/src/hid/nelma/nelma.c
@@ -704,6 +704,10 @@ nelma_set_layer(const char *name, int group, int empty)
 	int             idx = (group >= 0 && group < max_layer) ?
 	PCB->LayerGroups.Entries[group][0] : group;
 
+	if (SL_TYPE (idx) == SL_FINISHED) {
+		return 0;
+	}
+
 	if (name == 0) {
 		name = PCB->Data->Layer[idx].Name;
 	}
diff --git a/src/hid/png/png.c b/src/hid/png/png.c
index 0aa62ac..5390f29 100644
--- a/src/hid/png/png.c
+++ b/src/hid/png/png.c
@@ -823,6 +823,10 @@ png_set_layer (const char *name, int group, int empty)
   int idx = (group >= 0
 	     && group <
 	     max_layer) ? PCB->LayerGroups.Entries[group][0] : group;
+
+  if (SL_TYPE (idx) == SL_FINISHED)
+    return 0;
+
   if (name == 0)
     name = PCB->Data->Layer[idx].Name;
 
diff --git a/src/hid/ps/eps.c b/src/hid/ps/eps.c
index 15febe8..be49202 100644
--- a/src/hid/ps/eps.c
+++ b/src/hid/ps/eps.c
@@ -313,6 +313,10 @@ eps_set_layer (const char *name, int group, int empty)
   int idx = (group >= 0
 	     && group <
 	     max_layer) ? PCB->LayerGroups.Entries[group][0] : group;
+
+  if (SL_TYPE (idx) == SL_FINISHED)
+    return 0;
+
   if (name == 0)
     name = PCB->Data->Layer[idx].Name;
 
diff --git a/src/hid/ps/ps.c b/src/hid/ps/ps.c
index d742a5f..cb96675 100644
--- a/src/hid/ps/ps.c
+++ b/src/hid/ps/ps.c
@@ -516,6 +516,10 @@ ps_set_layer (const char *name, int group, int empty)
   int idx = (group >= 0
 	     && group <
 	     max_layer) ? PCB->LayerGroups.Entries[group][0] : group;
+
+  if (SL_TYPE (idx) == SL_FINISHED)
+    return 0;
+
   if (name == 0)
     name = PCB->Data->Layer[idx].Name;
